ocaml/typing/env.ml

```diff
@@ -1524,16 +1560,18 @@ let add_to_tbl id decl tbl =
   let decls = try NameMap.find id tbl with Not_found -> [] in
   NameMap.add id (decl :: decls) tbl
 
+exception Primitives_don't_have_addresses
+
 let value_declaration_address (_ : t) id decl =
   match decl.val_kind with
-  | Val_prim _ -> EnvLazy.create_failed Not_found
-  | _ -> EnvLazy.create_forced (Aident id)
+  | Val_prim _ -> EnvLazy.create_failed Primitives_don't_have_addresses
```

Comment:
Why the change in exception?

Reply:
I should have written it down, but as I recall, there was something that was tricky to debug because the `Not_found` was caught. I suppose `Fatal_error "Primitives don't have addresses"` would be better, since we don't intend that anyone catch the exception.

Reply:
Changed to `Invalid_argument "Primitives don't have addresses"`.